[docker]vcpkg install returns "A suitable version of git was not found (required v2.7.4)"

**Describe the bug**
After a fresh install of vcpkg I run `./vcpkg install boost` and get the following output:

```
A suitable version of git was not found (required v2.7.4) and unable to automatically download a portable one. Please install a newer version of git.
```
My git version is fine:
```
bash-4.2# git --version
git version 2.23.1
```

The actual problem appears to be that no URL is provided for a git tool download in the [scripts/vcpkgTools.xml](https://github.com/microsoft/vcpkg/blob/master/scripts/vcpkgTools.xml) file and since I am not using the `-useSystemBinaries` when bootstrapping it never attempts my local git version in [toolsrc/src/vcpkg/tools.cpp](https://github.com/microsoft/vcpkg/blob/master/toolsrc/src/vcpkg/tools.cpp) because the git provider has no fallback.

Given vcpkg itself is downloaded using git could we not at least have a fallback to the git in path? The version check would still take place. This may make sense to add as a generic fallback to all of the providers in the case a local version cannot be downloaded. 

I've temporarily fudged this by running the following command: 

```
sed -i 's/get_path(paths, GitProvider())/{\"git\", \"0\"}/g' toolsrc/src/vcpkg/tools.cpp
```
which forces the use of my local git version.

**Environment**
- OS: Amazon Linux 2 (RHEL 7 equivalent)
- Compiler: GNU 7.3.1

**To Reproduce**
Steps to reproduce the behavior:
1. git clone https://github.com/Microsoft/vcpkg.git
2. cd vcpkg
3. ./bootstrap-vcpkg.sh
4. ./vcpkg integrate install
5. ./vcpkg install boost
6. See error

**Expected behavior**
Boost libraries should install 

**Failure logs** 
```
The following packages will be built and installed:
    boost[core]:x64-linux
  * boost-accumulators[core]:x64-linux
  * boost-algorithm[core]:x64-linux
  * boost-align[core]:x64-linux
  * boost-any[core]:x64-linux
  * boost-array[core]:x64-linux
  * boost-asio[core]:x64-linux
  * boost-assert[core]:x64-linux
  * boost-assign[core]:x64-linux
  * boost-atomic[core]:x64-linux
  * boost-beast[core]:x64-linux
  * boost-bimap[core]:x64-linux
  * boost-bind[core]:x64-linux
  * boost-build[core]:x64-linux
  * boost-callable-traits[core]:x64-linux
  * boost-chrono[core]:x64-linux
  * boost-circular-buffer[core]:x64-linux
  * boost-compatibility[core]:x64-linux
  * boost-compute[core]:x64-linux
  * boost-concept-check[core]:x64-linux
  * boost-config[core]:x64-linux
  * boost-container[core]:x64-linux
  * boost-container-hash[core]:x64-linux
  * boost-context[core]:x64-linux
  * boost-contract[core]:x64-linux
  * boost-conversion[core]:x64-linux
  * boost-convert[core]:x64-linux
  * boost-core[core]:x64-linux
  * boost-coroutine[core]:x64-linux
  * boost-coroutine2[core]:x64-linux
  * boost-crc[core]:x64-linux
  * boost-date-time[core]:x64-linux
  * boost-detail[core]:x64-linux
  * boost-disjoint-sets[core]:x64-linux
  * boost-dll[core]:x64-linux
  * boost-dynamic-bitset[core]:x64-linux
  * boost-endian[core]:x64-linux
  * boost-exception[core]:x64-linux
  * boost-filesystem[core]:x64-linux
  * boost-flyweight[core]:x64-linux
  * boost-foreach[core]:x64-linux
  * boost-format[core]:x64-linux
  * boost-function[core]:x64-linux
  * boost-function-types[core]:x64-linux
  * boost-functional[core]:x64-linux
  * boost-fusion[core]:x64-linux
  * boost-geometry[core]:x64-linux
  * boost-gil[core]:x64-linux
  * boost-graph[core]:x64-linux
  * boost-graph-parallel[core]:x64-linux
  * boost-hana[core]:x64-linux
  * boost-heap[core]:x64-linux
  * boost-histogram[core]:x64-linux
  * boost-hof[core]:x64-linux
  * boost-icl[core]:x64-linux
  * boost-integer[core]:x64-linux
  * boost-interprocess[core]:x64-linux
  * boost-interval[core]:x64-linux
  * boost-intrusive[core]:x64-linux
  * boost-io[core]:x64-linux
  * boost-iostreams[core]:x64-linux
  * boost-iterator[core]:x64-linux
  * boost-lambda[core]:x64-linux
  * boost-lexical-cast[core]:x64-linux
  * boost-local-function[core]:x64-linux
  * boost-locale[core]:x64-linux
  * boost-lockfree[core]:x64-linux
  * boost-log[core]:x64-linux
  * boost-logic[core]:x64-linux
  * boost-math[core]:x64-linux
  * boost-metaparse[core]:x64-linux
  * boost-modular-build-helper[core]:x64-linux
  * boost-move[core]:x64-linux
  * boost-mp11[core]:x64-linux
  * boost-mpl[core]:x64-linux
  * boost-msm[core]:x64-linux
  * boost-multi-array[core]:x64-linux
  * boost-multi-index[core]:x64-linux
  * boost-multiprecision[core]:x64-linux
  * boost-numeric-conversion[core]:x64-linux
  * boost-odeint[core]:x64-linux
  * boost-optional[core]:x64-linux
  * boost-outcome[core]:x64-linux
  * boost-parameter[core]:x64-linux
  * boost-phoenix[core]:x64-linux
  * boost-poly-collection[core]:x64-linux
  * boost-polygon[core]:x64-linux
  * boost-pool[core]:x64-linux
  * boost-predef[core]:x64-linux
  * boost-preprocessor[core]:x64-linux
  * boost-process[core]:x64-linux
  * boost-program-options[core]:x64-linux
  * boost-property-map[core]:x64-linux
  * boost-property-tree[core]:x64-linux
  * boost-proto[core]:x64-linux
  * boost-ptr-container[core]:x64-linux
  * boost-qvm[core]:x64-linux
  * boost-random[core]:x64-linux
  * boost-range[core]:x64-linux
  * boost-ratio[core]:x64-linux
  * boost-rational[core]:x64-linux
  * boost-regex[core]:x64-linux
  * boost-safe-numerics[core]:x64-linux
  * boost-scope-exit[core]:x64-linux
  * boost-serialization[core]:x64-linux
  * boost-signals2[core]:x64-linux
  * boost-smart-ptr[core]:x64-linux
  * boost-sort[core]:x64-linux
  * boost-spirit[core]:x64-linux
  * boost-stacktrace[core]:x64-linux
  * boost-statechart[core]:x64-linux
  * boost-static-assert[core]:x64-linux
  * boost-system[core]:x64-linux
  * boost-test[core]:x64-linux
  * boost-thread[core]:x64-linux
  * boost-throw-exception[core]:x64-linux
  * boost-timer[core]:x64-linux
  * boost-tokenizer[core]:x64-linux
  * boost-tti[core]:x64-linux
  * boost-tuple[core]:x64-linux
  * boost-type-erasure[core]:x64-linux
  * boost-type-index[core]:x64-linux
  * boost-type-traits[core]:x64-linux
  * boost-typeof[core]:x64-linux
  * boost-ublas[core]:x64-linux
  * boost-units[core]:x64-linux
  * boost-unordered[core]:x64-linux
  * boost-utility[core]:x64-linux
  * boost-uuid[core]:x64-linux
  * boost-variant[core]:x64-linux
  * boost-variant2[core]:x64-linux
  * boost-vcpkg-helpers[core]:x64-linux
  * boost-vmd[core]:x64-linux
  * boost-wave[core]:x64-linux
  * boost-winapi[core]:x64-linux
  * boost-xpressive[core]:x64-linux
  * boost-yap[core]:x64-linux
  * bzip2[core]:x64-linux
  * libiconv[core]:x64-linux
  * liblzma[core]:x64-linux
  * openssl[core]:x64-linux
  * openssl-unix[core]:x64-linux
  * zlib[core]:x64-linux
  * zstd[core]:x64-linux
Additional packages (*) will be modified to complete this operation.
Starting package 1/144: boost-vcpkg-helpers:x64-linux
Building package boost-vcpkg-helpers[core]:x64-linux...
A suitable version of git was not found (required v2.7.4) and unable to automatically download a portable one. Please install a newer version of git.
```

**Additional context**
I have successfully used vcpkg on other Amazon Linux distro's without issue when using the `-useSystemBinaries` when bootstrapping.

@LilyWangL if you are familiar with docker you can create the exact image of the server I am using with the following commands: 

```
git clone https://github.com/aws/aws-codebuild-docker-images.git
cd aws-codebuild-docker-images
cd al2/x86_64/standard/2.0
docker build -t aws/codebuild/amazonlinux2-x86_64-standard:2.0 .
docker run -it --entrypoint sh aws/codebuild/amazonlinux2-x86_64-standard:2.0 -c bash
```
then in that shell:

git clone https://github.com/Microsoft/vcpkg.git
cd vcpkg
./bootstrap-vcpkg.sh
./vcpkg integrate install
./vcpkg install boost
See error

Although the sed solution allowed me to continue I'm now hitting an error on any package with a patch:
```
Starting package 1/23: liblzma:x64-linux
Building package liblzma[core]:x64-linux...
-- Downloading https://github.com/xz-mirror/xz/archive/v5.2.4.tar.gz...
-- Extracting source /vcpkg/downloads/xz-mirror-xz-v5.2.4.tar.gz
-- Applying patch enable-uwp-builds.patch
CMake Error at scripts/cmake/vcpkg_apply_patches.cmake:52 (message):
  Applying patch failed.  Patch needs to be updated to work with source being
  used by vcpkg!
Call Stack (most recent call first):
  scripts/cmake/vcpkg_extract_source_archive_ex.cmake:134 (vcpkg_apply_patches)
  scripts/cmake/vcpkg_from_github.cmake:139 (vcpkg_extract_source_archive_ex)
  ports/liblzma/portfile.cmake:3 (vcpkg_from_github)
  scripts/ports.cmake:90 (include)


Error: Building package liblzma:x64-linux failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: liblzma:x64-linux
  Vcpkg version: 2020.02.04-unknownhash

Additionally, attach any relevant sections from the log files above.
```

 so I assume git is still not being found correctly when this runs also. 

I've since added cmake and ninja to the server manually and symlinked these to `/usr/bin` and although cmake, ninja and git can now be found in the path, when I pass `-useSystemBinaries` during bootstrap and try and run `./vcpkg install liblzma` for example it, still downloads cmake, ninja and reports the git error  again as if the `VCPKG_FORCE_SYSTEM_BINARIES` environment variable is not there. 

Any ideas?

--- a/toolsrc/src/vcpkg/base/files.cpp
+++ b/toolsrc/src/vcpkg/base/files.cpp
@@ -666,6 +666,10 @@ namespace vcpkg::Files
 #if defined(_WIN32)
             static constexpr StringLiteral EXTS[] = {".cmd", ".exe", ".bat"};
             auto paths = Strings::split(System::get_environment_variable("PATH").value_or_exit(VCPKG_LINE_INFO), ";");
+#else
+            static constexpr StringLiteral EXTS[] = {""};
+            auto paths = Strings::split(System::get_environment_variable("PATH").value_or_exit(VCPKG_LINE_INFO), ":");
+#endif
 
             std::vector<fs::path> ret;
             std::error_code ec;
@@ -684,16 +688,6 @@ namespace vcpkg::Files
             }
 
             return ret;
-#else
-            const std::string cmd = Strings::concat("which ", name);
-            auto out = System::cmd_execute_and_capture_output(cmd);
-            if (out.exit_code != 0)
-            {
-                return {};
-            }
-
-            return Util::fmap(Strings::split(out.output, "\n"), [](auto&& s) { return fs::path(s); });
-#endif
         }
     };
 

